File: src/main/java/me/arasple/mc/trhologram/TrHologram.java
package me.arasple.mc.trhologram;

import io.izzel.taboolib.loader.Plugin;
import io.izzel.taboolib.module.config.TConfig;
import io.izzel.taboolib.module.inject.TInject;
import io.izzel.taboolib.module.locale.TLocale;
import io.izzel.taboolib.module.locale.logger.TLogger;
import me.arasple.mc.trhologram.hologram.HologramManager;
import me.arasple.mc.trhologram.utils.FileWatcher;

/**
 * @author Arasple
 * @date 2020/1/1 16:57
 */
@Plugin.Version(5.17)
public final class TrHologram extends Plugin {

    @TInject(value = "settings.yml", locale = "LOCALE")
    public final static TConfig SETTINGS = null;
    @TInject("§2Tr§aHologram")
    public final static TLogger LOGGER = null;
    @TInject(state = TInject.State.LOADING, init = "init", active = "active")
    private final static TrHologramLoader LOADER = null;

    @Override
    public void onStopping() {
        FileWatcher.INSTANCE.getWatcher().unregisterAll();
        HologramManager.INSTANCE.forceWrite();
        TLocale.sendToConsole("PLUGIN.DISABLED");
    }

    public static double getTrVersion() {
        return 1.1;
    }

}

File: src/main/kotlin/me/arasple/mc/trhologram/TrHologram.kt
package me.arasple.mc.trhologram

import io.izzel.taboolib.Version
import io.izzel.taboolib.loader.Plugin
import io.izzel.taboolib.loader.PluginBoot
import io.izzel.taboolib.module.locale.TLocale
import me.arasple.mc.trhologram.api.Settings
import me.arasple.mc.trhologram.module.conf.HologramLoader
import me.arasple.mc.trhologram.module.display.Hologram
import org.bukkit.Bukkit


/**
 * @author Arasple
 * @date 2021/1/25 12:11
 */
object TrHologram : Plugin() {


    override fun onLoad() {
        TLocale.sendToConsole("Plugin.Loading", Bukkit.getBukkitVersion())
    }

    override fun onEnable() {
        if (Version.isBefore(Version.v1_9)) {
            PluginBoot.setDisabled(true)
            return
        }

        Settings.init()
        HologramLoader.load(Bukkit.getConsoleSender())
        TLocale.sendToConsole("Plugin.Enabled", plugin.description.version)
    }

    override fun onDisable() {
        Hologram.holograms.forEach(Hologram::destroy)
    }

}